analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/RoslynSymbolicExecution.cs

```diff
@@ -142,6 +157,14 @@ context.Operation.Instance.Kind switch
                 _ => context.State
             };
 
+        private ProgramState CleanStateAfterBlock(ProgramState programState, BasicBlock block)
+        {
+            var liveVariables = lva.LiveOut(block).Union(nonInDeclarationParameters); // LVA excludes out and ref parameters
```

Comment:
```suggestion
            var liveVariables = lva.LiveOut(block).Concat(nonInDeclarationParameters).ToHashSet(); // LVA excludes out and ref parameters
```

Reply:
`ToHashSet` because of the `.Contains` usage below. It can be invoked many times on a bigger set.
`Concat` because it will not make sense to use `Union` after `ToHashSet` to avoid hashing it twice.